nixos/modules/services/misc/guix/default.nix

```
@@ -0,0 +1,325 @@
+
+  options.services.guix = with lib; {
+    enable = mkEnableOption "Guix build daemon service";
+
+    group = mkOption {
+      type = types.str;
+      default = "guixbuild";
+      example = "guixbuild";
+      description = ''
+        The group of the Guix build user pool.
+      '';
+    };
+
+    userPrefix = mkOption {
+      type = types.str;
+      default = "guixbuilder";
+      example = "guixbuilder";
+      description = ''
+        The name prefix for the Guix build user pool.
+      '';
+    };
+
+    nrBuildUsers = mkOption {
+      type = types.int;
+      description = ''
+        Number of Guix build users to be used in the build pool.
+      '';
+      default = 10;
+      example = 20;
+    };
+
+    extraArgs = mkOption {
+      type = with types; listOf str;
+      default = [ ];
+      example = [ "--max-jobs=4" "--debug" ];
+      description = ''
+        Extra flags to pass to the Guix daemon service.
+      '';
+    };
+
+    package = mkPackageOption pkgs "guix" {
+      extraDescription = ''
+        It should contain {command}`guix-daemon` and {command}`guix` executable.
+      '';
+    };
+
+    storeDir = mkOption {
+      type = types.str;
+      default = "/gnu/store";
+      defaultText = "/gnu/store";
+      description = ''
+        The store directory where the Guix service will serve to/from. Take
+        note Guix cannot take advantage of substitutes if you set it other than
+        {path}`/gnu/store` since most of the cached builds are
+        assumed in there.
+
+        ::: {.warning}
+        This will also recompile the package with the specified option so you
+        better have a good reason to change it.
+        :::
+      '';
+    };
+
+    stateDir = mkOption {
+      type = types.str;
+      default = "/var";
+      defaultText = "/var";
+      description = ''
+        The state directory where Guix service will store its data such as its
+        user-specific profiles, cache, and state files.
+
+        ::: {.warning}
+        Changing it other than the default will rebuild the package.
+        :::
+      '';
+      example = "/gnu/var";
+    };
+
+    publish = {
+      enable = mkEnableOption "substitute server for your Guix store directory";
+
+      port = mkOption {
+        type = types.port;
+        default = 8181;
+        example = 8200;
+        description = ''
+          Port of the substitute server to listen to.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "nobody";
+        defaultText = "nobody";
+        description = ''
+          Name of the user to change once the server is up.
+        '';
+      };
+
+      extraArgs = mkOption {
+        type = with types; listOf str;
+        description = ''
+          Extra flags to pass to the substitute server.
+        '';
+        default = [];
+        example = lib.literalExpression ''
+          [ "--compression=zstd:6" "--repl" "--discover=no" ]
+        '';
```

Examples without references that might evaluate unintentionally (like `pkgs.*`, `lib.*`, `config.*`, `options.*`) can just stay nonliteral.

```suggestion
        example = [ "--compression=zstd:6" "--repl" "--discover=no" ]
```

---

Got it.